/*

Copyright (c) 2013, Coolplay (pengchong)
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

-- Redistributions of source code must retain the above copyright notice, this
list of conditions and the following disclaimer.

-- Redistributions in binary form must reproduce the above copyright notice, this
list of conditions and the following disclaimer in the documentation and/or
other materials provided with the distribution.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

*/

package json_protobuf.test;


message employee {
        required uint32 id = 1;
        required string first_name = 2;
        required string last_name = 3;

        optional string date_of_birth = 4;

        enum pay_type_enum {
             hourly = 1;
             salaried = 2;
        }

        optional pay_type_enum pay_type = 5;

        optional uint32 salary = 6;
        optional uint32 hourly_wage = 7;

        optional bool active = 8;

        repeated string departments = 9;

        optional uint64 hire_date = 10;
}

message A {
        required uint32 a = 1;
}

message B {
        required uint32 a = 1;
        required uint32 b = 2;
        required A a_message = 3;
}

message C {
        required uint32 a = 1;
        required uint32 b = 2;
        required uint32 c = 3;
        required A a_message = 4;
        required B b_message = 5;
}
